src/main/java/com/deavensoft/paketomat/courier/CourierServiceImpl.java

```diff
@@ -22,76 +23,87 @@ public class CourierServiceImpl implements CourierService {
     private final CenterService centerService;
     private final CityService cityService;
     private final DispatcherService dispatcherService;
-    private List<Package> packagesToDispatch = new ArrayList<>();
-    private List<City> citiesToDispatch = new ArrayList<>();
+    private final List<Package> packagesToDispatch = new ArrayList<>();
+    private final List<City> citiesToDispatch = new ArrayList<>();
 
-    public List<Courier> findAllCouriers(){
+    public List<Courier> findAllCouriers() {
         return courierRepository.findAll();
     }
 
-    public void saveCourier(Courier newCourier){
+    public void saveCourier(Courier newCourier) {
         courierRepository.save(newCourier);
     }
 
 
-    public Optional<Courier> getCourierById(Long id){
+    public Optional<Courier> getCourierById(Long id) {
         return courierRepository.findById(id);
     }
 
-    public void deleteCourierById(Long id){
+    public void deleteCourierById(Long id) {
         courierRepository.deleteById(id);
     }
+
     @Override
     public List<Package> getPackagesForCourier(String city) throws PaketomatException {
         getPackagesToDispatch();
         findCitiesInRadius(city);
-        return filterPackagesToDispatch();
+        return deliverPackageInPaketomat(filterPackagesToDispatch(packagesToDispatch,citiesToDispatch));
     }
 
-    public void getPackagesToDispatch() {
+    public List<Package> getPackagesToDispatch() {
         List<Package> packageList = centerService.getAllPackages();
         packagesToDispatch.clear();
 
-        for(Package p : packageList){
-            if(p.getStatus().equals(Status.TO_DISPATCH)) {
+        for (Package p : packageList) {
+            if (p.getStatus().equals(Status.TO_DISPATCH)) {
                 packagesToDispatch.add(p);
             }
         }
         log.info("List with packages TO_DISPATCH is made");
+        return packageList;
     }
 
-    public void findCitiesInRadius(String city) throws PaketomatException {
+    public List<City> findCitiesInRadius(String city) throws PaketomatException {
         double maxDistance = 100.0;
         double distance;
         List<City> citiesList = cityService.getAllCities();
         citiesToDispatch.clear();
 
-        for(City c : citiesList){
+        for (City c : citiesList) {
             distance = dispatcherService.findDistance(city, c.getName());
-            if(distance <= maxDistance){
+            if (distance <= maxDistance) {
                 citiesToDispatch.add(c);
             }
         }
         log.info("List with cities in 100km radius from city " + city + " is made");
+        return citiesList;
     }
 
-    public List<Package> filterPackagesToDispatch(){
+    public List<Package> filterPackagesToDispatch(List<Package> packagesToDispatch, List<City> citiesToDispatch ) {
```

Comment:
Unused method parameters. 

- [ ] Fix it

Reply:
Fixed

Reply:
What is fixed here?

Reply:
Check again :)